File: docs/03-conjunction/ecmascript/executor.mjs
import { conjunction } from './propositional_logic.mjs';

let p = true;
let q = true;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: true

p = true;
q = false;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: false

p = false;
q = true;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: false

p = false;
q = false;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: false
    
p = undefined;
q = false;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: undefined

p = undefined;
q = true;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: undefined

p = true;
q = undefined;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: undefined

p = false;
q = undefined;
console.log('(p='+p+' conjunctionOperator q='+q+') = '+conjunction(p, q)); // Output: undefined

File: docs/03-conjunction/python3/executor.py
# python3 executor.py
import propositional_logic

def main():

    p = True
    q = True
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: True

    p = True
    q = False
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: False

    p = False
    q = True
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: False

    p = False
    q = False
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: False

    p = None
    q = False
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: None

    p = None
    q = True
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: None

    p = True
    q = None
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: None

    p = False
    q = None
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: None

    p = None
    q = None
    print('(p=',p,' conjunctionOperator q=',q,') = ',propositional_logic.conjunction(p, q), sep='') # Output: None


if __name__ == "__main__":
    main()

